cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -900,3 +903,462 @@ class _SynchronousMemoryResource(MemoryResource):
     @property
     def device_id(self) -> int:
         return self._dev_id
+
+@dataclass
+class VMMAllocationOptions:
+    """A configuration object for the VMMAllocatedMemoryResource
+       Stores configuration information which tells the resource how to use the CUDA VMM APIs
+    """
+    """
+    Configuration for CUDA VMM allocations.
+
+    Args:
+        handle_type: Export handle type for the physical allocation. Use
+            CU_MEM_HANDLE_TYPE_POSIX_FILE_DESCRIPTOR on Linux if you plan to
+            import/export the allocation (required for cuMemRetainAllocationHandle).
+            Use CU_MEM_HANDLE_TYPE_NONE if you don't need an exportable handle.
+        gpu_direct_rdma: Hint that the allocation should be GDR-capable (if supported).
+        granularity: 'recommended' or 'minimum'. Controls granularity query and size rounding.
+        addr_hint: A (optional) virtual address hint to try to reserve at. 0 -> let CUDA choose.
+        addr_align: Alignment for the VA reservation. If None, use the queried granularity.
+        peers: Extra device IDs that should be granted access in addition to `device`.
+        self_access: Access flags for the owning device ('rw', 'r', or 'none').
+        peer_access: Access flags for peers ('rw' or 'r').
+    """
+    # Human-friendly strings; normalized in __post_init__
+    allocation_type: str = "pinned"          # pinned
+    location_type: str = "device"            # device
+    handle_type: str = "posix-fd"           # posix-fd | generic | none
```

Comment:
based on my limited understanding in typing, we could make it more informative
```python
VMMHandleTypeT = Literal["posix-fd", "generic", "none"]

@dataclass
class VMMAllocationOptions:
   ...
   handle_type: VMMHandleTypeT = "posix_fd"
```

Reply:
Did not know about that feature. Will do.